.github/workflows/weekly-preview.yml

```diff
@@ -0,0 +1,53 @@
+name: weekly-preview
+
+on:
+  # TODO: re-enable
+  # schedule:
+  # - cron: "0 2 * * 0"  # 02:00 of every Sunday
+  # for testing on this branch; TODO: remove
+  push:
+    branches:
+      - aihsani/weekly-build
+
+jobs:
+  packaging:
+    if: github.repository == 'Project-MONAI/MONAILabel'
+    runs-on: ubuntu-latest
+    env:
+      DEV_RELEASE_VERSION: 0.1
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        ref: main
+        fetch-depth: 0
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version:  3.8
+    - name: Install setuptools
+      run: |
+        python -m pip install --user --upgrade setuptools wheel
+    - name: Build distribution
+      run: |
+        export HEAD_COMMIT_ID=$(git rev-parse HEAD)
+        sed -i 's/name\ =\ monai-label$/name\ =\ monai-label-weekly/g' setup.cfg
+        echo "__commit_id__ = \"$HEAD_COMMIT_ID\"" >> monailabel/__init__.py
+        git diff setup.cfg monailabel/__init__.py
+        git config user.name "CI Builder"
+        git config user.email "monai.contact@gmail.com"
+        git add setup.cfg monailabel/__init__.py
+        git commit -m "Weekly build at $HEAD_COMMIT_ID"
+        export YEAR_WEEK=$(date +'%y%U')
+        echo "Year week for tag is ${YEAR_WEEK}"
+        if ! [[ $YEAR_WEEK =~ ^[0-9]{4}$ ]] ; then echo "Wrong 'year week' format.  Should be 4 digits."; exit 1 ; fi
+        git tag "$DEV_RELEASE_VERSION.dev${YEAR_WEEK}"
+        git log -1
+        git tag --list
+        python setup.py sdist bdist_wheel
+
```

Comment:
I think adding a section like this:
```
    - name: Upload artifacts
      uses: actions/upload-artifact@v1
      with:
        name: dist
        path: dist/
```
we'll be able to get the `dist` folder via the github UI https://docs.github.com/en/actions/managing-workflow-runs/downloading-workflow-artifacts